mdagent/tools/base_tools/analysis_tools/inertia.py

```diff
@@ -0,0 +1,159 @@
+        for i, tensor in enumerate(inertia_tensors):
+            eigenvalues, _ = np.linalg.eigh(tensor)
+            principal_moments[i] = eigenvalues
+
+        self.moments_of_inertia = principal_moments
+        self.min_moi = np.min(self.moments_of_inertia)  # min of all frames & moments
+        self.avg_moi = np.mean(
+            self.moments_of_inertia
+        )  # average of all frames & moments
+
+        # save to file
+        file_id = f"MOI_{self.mol_name}"
+        description = (f"Moments of inertia tensor for {self.mol_name}",)
+        csv_path = save_to_csv(
+            self.path_registry, self.moments_of_inertia, file_id, description
+        )
+        message = (
+            f"Average Moment of Inertia Tensor: {self.avg_moi}, "
+            f"Data saved to: {csv_path} with file ID {file_id}"
+        )
+        return message
+
+    def plot_moi(self):
+        """
+        Analyze and visualize the principal moments of inertia.
+        """
+        message = ""
+        if self.moments_of_inertia is None:
+            message += self.calculate_moment_of_inertia()
+
+        if len(self.moments_of_inertia) == 1:  # only one frame
+            message += "Only one frame in trajectory, no plot generated."
+            return message
+
+        fig_analysis = f"MOI_{self.mol_name}"
+        fig_name = self.path_registry.write_file_name(
+            type=FileType.FIGURE,
+            fig_analysis=fig_analysis,
+            file_format="png",
+        )
+        fig_id = self.path_registry.get_fileid(file_name=fig_name, type=FileType.FIGURE)
+
+        plt.plot(self.avg_moi, label="$I_avg$", linestyle="--", color="black")
+        plt.plot(self.min_moi, label="$I_min$", linestyle="--", color="red")
+        plt.plot(self.moments_of_inertia[:, 0], label="$I_1$")  # smallest MOI
+        plt.plot(self.moments_of_inertia[:, 1], label="$I_2$")
+        plt.plot(self.moments_of_inertia[:, 2], label="$I_3$")  # largest MOI
+        plt.xlabel("Frame")
+        plt.ylabel("Moments of Inertia")
+        plt.title("Moments of Inertia over Time")
+        plt.legend()
+        plt.savefig(f"{self.path_registry.ckpt_figures}/{fig_name}")
+        plt.close()
+        self.path_registry.map_path(
+            fig_id,
+            f"{self.path_registry.ckpt_figures}/{fig_name}",
+            description=f"Plot of moments of inertia over time for {self.mol_name}",
+        )
+        message += (
+            f"Plot of moments of inertia over time saved as: "
+            f"{fig_name}.png with plot ID {fig_id}. "
+        )
+        return message
+
+
+class MomentOfInertiaToolInput(BaseModel):
+    top_fileid: str = Field(None, description="File ID for the topology file.")
+    traj_fileid: Optional[str] = Field(
+        None, description="File ID for the trajectory file."
+    )
+    molecule_name: Optional[str] = Field(
+        None, description="Name of the molecule or protein."
+    )
```

Comment:
Not sure we need to use this for this functionality, seems overly complex.

Reply:
you may be actually right. I removed them